hanadb_exporter/main.py

```diff
@@ -83,18 +108,13 @@ def run():
         setup_logging(config)
     else:
         logging.basicConfig(level=args.verbosity or logging.INFO)
-   
+
     logger = logging.getLogger(__name__)
     metrics = args.metrics
     connector = hdb_connector.HdbConnector()
+
     try:
-        hana_config = config['hana']
-        connector.connect(
-            hana_config['host'],
-            hana_config.get('port', 30015),
-            user=hana_config['user'],
-            password=hana_config['password']
-        )
+        connect(connector, config, logger)
```

Comment:
do we need the `logger` as argument? or could we just delegate the logging init to the specif function without passing it to arg function?

Reply:
so having `connect(connector, config`) should looks more natural to me for reading it.

Logger it seems more constructs we pass around but we could spare

Reply:
I could that
```
logger = logging.getLogger(__name__)
```
in the `connect` method, but if for some reason we change `__name__` by something else the logger would be different.

Other option is to initialize the `logger` at the top of the file as a global parameter and use it in the file,